### ISSUE TITLE
Add Connection::setPdo()?

### ISSUE THREAD

**Issue by burzum** [laravel/framework#25663]
I'm trying to write an adapter for an authentication library project I'm working on. To be able to lookup an user from the DB using the Eloquent ORM I need to use a model and to test it I need to somehow hook it up with a connection to my phpunit/dbunit connection instance, which is a simple PDO object.

I've spend now some time without success digging in the ORM code of Laravel / Eloquent without success. I've seen that you can get the PDO instance from a connection but I can't set it.

```php
 $this->capsule = new Capsule;
// This is obviously not working because it takes an array config
$this->capsule->addConnection($this->database->pdo());
 $this->capsule->setAsGlobal();
$this->capsule->bootEloquent();
```

Is there any way to set an existing PDO instance to the connection or any other way I could archive the desired use case of using my PDO sqlite in memory connection for my unit tests with Eloquent?

Other references:

* https://laracasts.com/discuss/channels/general-discussion/how-to-construct-illuminatedatabase-capsule-with-existing-pdo-connection
* https://stackoverflow.com/questions/35316190/use-existing-pdo-link-in-laravel

**Comment by mfn** [comment]
@burzum mind to share? 😄

**Comment by mfn** [comment]
Ah, custom `ConnectionResolverInterface` on the model => thx!

**Comment by burzum** [comment]
@mfn yes the ticket was misleading, the actual problem was to set the connection object, not the PDO instance to the connection. There is already setPdo(), no idea why I've missed that in the first place. However, it would be nice if there would be a method on the model that allows me to simply set the connection.
